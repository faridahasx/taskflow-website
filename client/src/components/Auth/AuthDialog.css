#auth-container {
    width: 90vw;
    max-width: 500px;
    min-width: 200px;
    height: calc((var(--vh) / 100) * 90);
    max-height: 700px;
    background-color: white;
    overflow: auto;
    border-radius: 10px;
    box-shadow: var(--box-shadow);
    align-items: center;
}

#auth-container::-webkit-scrollbar-track {
    margin: 5px 0;
}

#auth-top {
    width: 100%;
    padding: 5px;
}

#auth-top h1 {
    width: 100%;
    font-size: 23px;
}


#auth-wrapper {
    width: 100%;
    padding: 20px 10%;
    align-items: center;
}

#change-auth-method-btn,
#google-btn {
    font-weight: 600;
}

#change-auth-method-btn:hover,
#google-btn:hover {
    color: rgb(85, 85, 85);
}


#change-auth-method-btn {
    width: 100%;
}