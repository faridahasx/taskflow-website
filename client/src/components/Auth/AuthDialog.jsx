// External imports
import { useState } from "react";
import PropTypes from "prop-types";
// Components
import AuthFormContainer from "../../containers/AuthFormContainer";
import CloseButton from "../IconButtons/CloseButton";
import Modal from "../Modal/Modal";
// Styles
import "./AuthDialog.css";

const signIn = {
  heading: "Sign in to continue",
  path: "login",
  subtmitButtonText: "Sign in",
  changeMethod: "Sign up instead",
};

const signUp = {
  heading: "Sign up to start",
  path: "register",
  subtmitButtonText: "Sign up",
  changeMethod: "Sign in instead",
};

const AuthDialog = ({ handleCloseAuthDialog }) => {
  const [authMethod, setAuthMethod] = useState(signIn);
  const { heading, changeMethod, path, subtmitButtonText } = authMethod;

  const handleChangeAuthMethod = () => {
    setAuthMethod(path === signIn.path ? signUp : signIn);
  };

  return (
    <Modal handleClose={handleCloseAuthDialog}>
      <div>
        <div id="auth-container" className="column">
          <div className="column" id="auth-top">
            <CloseButton onClick={handleCloseAuthDialog} />
            <h1 className="center">{heading}</h1>
          </div>
          <div id="auth-wrapper" className="column">
            <AuthFormContainer
              key={path}
              path={path}
              subtmitButtonText={subtmitButtonText}
            />
            <button
              onClick={handleChangeAuthMethod}
              id="change-auth-method-btn"
              className="center"
            >
              {changeMethod}
            </button>
          </div>
        </div>
      </div>
    </Modal>
  );
};

AuthDialog.propTypes = {
  handleCloseAuthDialog: PropTypes.func.isRequired,
};

export default AuthDialog;
