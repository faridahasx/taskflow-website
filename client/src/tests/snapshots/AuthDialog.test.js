import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import AuthDialog from "../../components/Auth/AuthDialog";
import mockInitialState, { mockStore } from "../mocks/mockReduxState";


test("AuthDialog snapshot", () => {
  const { asFragment } = render(
    <Provider store={mockStore(mockInitialState)}>
      <AuthDialog handleCloseAuthDialog={jest.fn()} />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot("AuthDialog");
});
