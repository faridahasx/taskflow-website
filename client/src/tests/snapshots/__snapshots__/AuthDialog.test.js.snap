// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthDialog snapshot: AuthDialog 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div>
      <div
        class="column"
        id="auth-container"
      >
        <div
          class="column"
          id="auth-top"
        >
          <button
            class="icon "
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
          <h1
            class="center"
          >
            Sign in to continue
          </h1>
        </div>
        <div
          class="column"
          id="auth-wrapper"
        >
          <form
            class="column"
            id="auth-form"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated auth-form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
              for="email"
            >
              Email:
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-q0jhri-MuiInputBase-root-MuiInput-root"
              required=""
            >
              <input
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="email"
                name="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated auth-form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
              for="password"
            >
              Password:
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-q0jhri-MuiInputBase-root-MuiInput-root"
              required=""
            >
              <input
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Show password"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <button
              class="center submit-btn"
              form="auth-form"
              title="Sign in"
              type="submit"
            >
              Sign in
            </button>
          </form>
          <span>
            Or
          </span>
          <button
            class="center"
            id="google-btn"
          >
            <span
              class="center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="GoogleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
                />
              </svg>
            </span>
            <span
              class="center"
            >
              Continue with Google
            </span>
          </button>
          <button
            class="center"
            id="change-auth-method-btn"
          >
            Sign up instead
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
